packages/flutter_tools/lib/src/build_system/targets/native_assets.dart

```diff
@@ -228,6 +250,40 @@ class NativeAssets extends Target {
     }
   }
 
+  List<AndroidArch> _androidArchs(
+    TargetPlatform targetPlatform,
+    String? androidArchsEnvironment,
+  ) {
+    switch (targetPlatform) {
```

Comment:
You have a few of these converters from Architecture to TargetPlatform to AndroidArch to string representations. Would it make sense to put them together in their own file with their own unit tests?

Reply:
+1

Reply:
@zanderso +1 to Reid or me?

Reply:
Sorry, GitHub only showed me Reid's comment. I was giving a +1 to the suggestion of avoiding duplicate code, where possible.